client/src/components/core/Modals/EditBoardModal.js

```diff
@@ -0,0 +1,165 @@
+// This is with reactstrap and bootstrap
+import {
+	Modal,
+	ModalBody,
+	FormGroup,
+	Label,
+	Form,
+	Input,
+	ModalHeader,
+} from "reactstrap";
+import "bootstrap/dist/css/bootstrap.min.css";
+import styles from "./Modals.module.css";
+import { Button } from "../Button";
+import { useState } from "react";
+import Image from "next/image";
+import cross from "@assets/icon-cross.svg";
+import { useMutation } from "@apollo/client";
+import { CREATE_BOARD, GET_BOARDS } from "../../../lib/graphql/queries";
+
+const EditBoardModal = ({
+	editBoardModalOpen,
+	setEditBoardModalOpen,
+	handleToggleEditBoard,
+}) => {
+	const [inputValue, setInputValue] = useState("");
+	const [boardName, setBoardName] = useState("");
+	const [boardColumns, setBoardColumns] = useState(["Todo", "Doing"]);
+	const [newBoard, setNewBoard] = useState({
+		boardName: boardName,
+		boardColumns: boardColumns,
+	});
+	const [createBoard] = useMutation(CREATE_BOARD, {
+		refetchQueries: [{ query: GET_BOARDS }],
+	});
+	const [errors, setErrors] = useState({
+		boardName: "",
+	});
+
+	const handleBoardNameChange = (e) => {
+		setInputValue(e.target.value);
+		const newBoardName = e.target.value;
+		setBoardName(newBoardName);
+	};
+
+	const handleBoardColumnChange = (e) => {
+		setInputValue(e.target.value);
+		boardColumns[e.target.id] = e.target.value;
+		setBoardColumns(boardColumns);
+	};
+
+	const handleEditColumnBtnClick = () => {
+		setBoardColumns([...boardColumns, ""]);
+	};
+
+	const handleXBtnClick = (e) => {
+		const newBoardColumns = [...boardColumns];
+		newBoardColumns.splice(e.target.id, 1);
+		setBoardColumns(newBoardColumns);
+	};
+
+	const handleCreateNewBoardSubmit = async (e) => {
+		// e.preventDefault();
+		if (boardName) {
+			setErrors({ boardName: "" });
+			await createBoard({ variables: { name: boardName } });
+			setEditBoardModalOpen(false);
+		} else {
+			setErrors({ boardName: "Board name is required" });
+		}
+	};
+	if (!editBoardModalOpen) return null;
+
+	return (
+		<div className={styles.modalOverlay}>
+			<div className={styles.modalContent}>
```

Comment:
If this is a modal, it's always good practice to use the native elements.  We also have this component https://github.com/Erikang20/Kanban-task-management-web-app/blob/main/client/src/components/core/Modals/Dialog.tsx which is intended to be reused (If possible), so we don't repeat ourselves. I noticed that we have a few of these modals around. I think for now it's fine as we are running out of time, but if we can we should avoid duplicating code ;)

Reply:
Ok sounds good! sorry I did leave my console logs to finish out the edit/delete board. Then I was going to delete them all .  I should've deleted them

Reply:
Nothing to be sorry about. It happens to us all ;)